Sources/StreamChatSwiftUI/ChatChannel/Reactions/ReactionUserView.swift

```diff
@@ -0,0 +1,155 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import SwiftUI
+
+/// View displaying single user reaction.
+struct ReactionUserView: View {
+    
+    @Injected(\.chatClient) private var chatClient
+    @Injected(\.fonts) private var fonts
+    
+    var reaction: ChatMessageReaction
+    var imageSize: CGFloat
+    
+    private var isCurrentUser: Bool {
+        chatClient.currentUserId == reaction.author.id
+    }
+    
+    private var authorName: String {
+        if isCurrentUser {
+            return L10n.Message.Reactions.currentUser
+        } else {
+            return reaction.author.name ?? reaction.author.id
+        }
+    }
+    
+    var body: some View {
+        VStack {
+            MessageAvatarView(
+                author: reaction.author,
+                size: CGSize(width: imageSize, height: imageSize),
+                showOnlineIndicator: false
+            )
+            .overlay(
+                VStack {
+                    Spacer()
+                    SingleReactionView(reaction: reaction)
+                        .frame(height: imageSize / 2)
+                }
+            )
+            
+            Text(authorName)
+                .multilineTextAlignment(.center)
+                .lineLimit(2)
+                .font(fonts.footnoteBold)
+                .frame(width: imageSize)
+        }
+        .padding(.vertical)
+        .padding(.horizontal, 8)
+    }
+}
+
+/// Helper view displaying the reaction overlay.
+struct SingleReactionView: View {
+    
+    @Injected(\.images) private var images
+    @Injected(\.colors) private var colors
+    @Injected(\.chatClient) private var chatClient
+    
+    var reaction: ChatMessageReaction
+    
+    private var isSentByCurrentUser: Bool {
+        reaction.author.id == chatClient.currentUserId
+    }
+    
+    private var backgroundColor: Color {
+        isSentByCurrentUser ? Color(colors.background) : Color(colors.background6)
+    }
+        
+    var body: some View {
+        VStack {
+            Spacer()
```

Comment:
One trick I learned to avoid the `Spacer()` (which is probably not really more efficient but gets rid of the construct, that feels weird to me)...
Is to give the other element a `.frame` modifier with `maxHeight: .infinity`. This way it spreads as large as possible and can be compared to `flex: 1` in CSS.

But just wanted to impress you with my knowledge here, perfectly fine code.

Reply:
Not sure it will work only with maxHeight - we want the view below the spacer always at the bottom. With the maxHeight approach it will be centered. I actually like the Spacer, it's very descriptive IMO.